### PR TITLE
Add new serialisable header

### LINKED ISSUE #184: Define and implement Message type
The `NetworkMessage` struct is a general container for all messages that are passed around the Tari network.

It comprises a `MessageHeader` which holds the message metadata including 
* message type
* source pub key
* destination pubkey <optional>
* encrypted?
* message signature

And the MessageBody, which is a byte array (having been serialised by MessagePack).

Child of #183

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description
Serialize and deserialze for the raw message into a message envelope 

Closes #184 
## Motivation and Context
The dispatcher needs to work with a deserialized message 

## Types of changes
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: infrastructure/comms/src/connection/message.rs [hunk 1]
```diff
@@ -118,4 +159,30 @@ mod test {
...
+    #[test]
+    fn test_ser_des() {
+        let version = 0;
+        let mut rng = rand::OsRng::new().unwrap();
+        let k = RistrettoSecretKey::random(&mut rng);
+        let p = RistrettoPublicKey::from_secret_key(&k);
+        let source = p;
+        let dest: NodeDestination<RistrettoPublicKey> = NodeDestination::Unknown;
+        let signature = vec![0];
+        let flags = 0; // IdentityFlags::ENCRYPTED;
```
Comment (CjS77) [inline infrastructure/comms/src/connection/message.rs]: I think you want `IdentityFlags::empty()` here

Comment (SWvheerden) [reply infrastructure/comms/src/connection/message.rs]: I actually wanted`IdentityFlags::ENCRYPTED` there, but I removed the flags struct in ln 66 for testing porposes. I never put it back. Hence the 0 there. But its fixed now.

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 2]
```diff
@@ -201,41 +201,45 @@ This illustrates the structure of a Tari message.
...
 | identity | 0     | 8               | `[u8;8]`  | The identifier that a `ZMQ_ROUTER` socket expects so that it knows the intended destination of the message. This can be thought of as a session token. |
 | version  | 1     | 1               | `u8`      | The wire protocol version.                                                                                                                             |
-| header   | 2     | varies          | `Vec<u8>` | Serialized bytes of data containing an unencrypted HeaderMessage.                                                                                      |
-| body     | 3     | varies          | `Vec<u8>` | Serialized bytes of data containing a unencrypted or encrypted [DataMessageBody].                                                                      |
+| header   | 2     | varies          | `Vec<u8>` | Serialized bytes of data containing an unencrypted MessageEnvelopeHeader.                                                                                      |
```
Comment (sdbondi) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
| header   | 2     | varies          | `Vec<u8>` | Serialized bytes of data containing an unencrypted [MessageEnvelopeHeader].                                                                                      |
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 3]
```diff
@@ -201,41 +201,45 @@ This illustrates the structure of a Tari message.
...
+| body     | 3     | varies          | `Vec<u8>` | Serialized bytes of data containing a unencrypted or encrypted [MessageEnvelopeBody].                                                                      |
 
 The header and decrypted body MUST be deserializable as per [RFC-0171: Message Serialisation](RFC-0171_MessageSerialisation.md)
 
-#### DataMessageHeader
+#### MessageEnvelope Header
```
Comment (sdbondi) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
#### MessageEnvelopeHeader
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 4]
```diff
@@ -697,12 +701,11 @@ If the HWM is hit:
 [communication node]: Glossary.md#communication-node
 [connectioncontext]: #connectioncontext
 [controlservice]: #controlservice
-[datamessage]: #datamessage
-[datamessagebody]: #datamessagebody
+[MessageEnvelope]: #MessageEnvelope
+[MessageEnvelopebody]: #MessageEnvelopebody
```
Comment (sdbondi) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]: Add `[MessageEnvelopeHeader](#MessageEnvelopeHeader]`

**Pull request merged**